### PR TITLE
Update SoftDeleteable to work with Doctrine 3.1

### PR DESCRIPTION
**Pull request by elfantome**
Update Gedmo\SoftDeleteable\Filter\SoftDeleteableFilter::addFilterConstraint signature to be compatible with Doctrine\ORM\Query\Filter\SQLFilter::addFilterConstraint

### CODE REVIEW

**Comment by mbabker** [comment]
Unfortunately this change breaks compatibility with ORM 2.x.

```sh
Fatal error: Declaration of Gedmo\SoftDeleteable\Filter\SoftDeleteableFilter::addFilterConstraint(Doctrine\ORM\Mapping\ClassMetadata $targetEntity, string $targetTableAlias): string must be compatible with Doctrine\ORM\Query\Filter\SQLFilter::addFilterConstraint(Doctrine\ORM\Mapping\ClassMetadata $targetEntity, $targetTableAlias) in /src/SoftDeleteable/Filter/SoftDeleteableFilter.php on line 54
```

**Comment by elfantome** [comment]
> Unfortunately this change breaks compatibility with ORM 2.x.
> 
> ```shell
> Fatal error: Declaration of Gedmo\SoftDeleteable\Filter\SoftDeleteableFilter::addFilterConstraint(Doctrine\ORM\Mapping\ClassMetadata $targetEntity, string $targetTableAlias): string must be compatible with Doctrine\ORM\Query\Filter\SQLFilter::addFilterConstraint(Doctrine\ORM\Mapping\ClassMetadata $targetEntity, $targetTableAlias) in /src/SoftDeleteable/Filter/SoftDeleteableFilter.php on line 54
> ```

In this case, maybe bumping the version number with a new release can solve the problem?

**Comment by mbabker** [comment]
I just looked at the install stats.  It is way too early to consider dropping ORM 2.x support, and the volunteer resources don't exist to support two major version branches of this package.

There might be a way to change the signature in a way to support both versions (like only adding the return type without the param type), but based on numbers and resources only supporting ORM 3.x right now is not the right decision.

**Comment by elfantome** [comment]
Makes sense.
I update my PR to only add a return type to the method to keep compatibility with ORM 2

**Review by mbabker — APPROVED** [review]

File: src/SoftDeleteable/Traits/SoftDeleteableEntity.php [hunk 1]
```diff
@@ -23,8 +23,6 @@
 trait SoftDeleteableEntity
 {
     /**
-     * @ORM\Column(type="datetime", nullable=true)
```
Comment (Seb33300) [inline src/SoftDeleteable/Traits/SoftDeleteableEntity.php:26]: Needs to be reverted

**Review by franmomu — APPROVED** [review]

**Comment by franmomu** [comment]
thanks @elfantome!

**Pull request merged**